shub/deploy.py

```diff
@@ -140,17 +141,17 @@ def _upload_egg(target, eggpath, project, version, auth):
     return make_deploy_request(url, data, files, auth)
 
 
-def _build_egg():
+def _build_egg(setup_file):
     closest = closest_scrapy_cfg()
     os.chdir(os.path.dirname(closest))
-    if not os.path.exists('setup.py'):
+    if not os.path.exists(setup_file):
```

Comment:
So, I think we should consider if we really want this feature.

I understand we needed it for an internal project with weird packaging and this change doesn't look like much, but the more arguments we add, the more decisions we are asking for our users to make.

IMO, it'd be better to fix the packaging of the internal project, keep shub's flow more straightforward and if some other user asks for this we consider adding it again.